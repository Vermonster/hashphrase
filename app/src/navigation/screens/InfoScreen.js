import React from 'react';
import { View } from 'react-native';
import { Card, Text } from 'react-native-paper';

const InfoScreen = () => (
  <View>
    <Card>
      <Text>Test card and text</Text>
    </Card>
  </View>
);

export default InfoScreen;
