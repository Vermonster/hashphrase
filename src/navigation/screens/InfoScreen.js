import React from 'react';
import { View } from 'react-native';
import { createStackNavigator } from 'react-navigation';
import { Card, Text } from 'react-native-paper';

const InfoScreen = () => (
  <View>
    <Card>
      <Text>Test card and text</Text>
    </Card>
  </View>
);

export default createStackNavigator({
  Info: InfoScreen,
});
