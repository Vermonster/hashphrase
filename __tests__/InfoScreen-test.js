import React from 'react';
import { shallow } from 'enzyme';
import { View } from 'react-native';
import { Card, Text } from 'react-native-paper';
import InfoScreen from '../src/navigation/screens/InfoScreen';

// currently a placeholder component for future expert tips/info screen
describe('<InfoScreen />', () => {
  it('should render the same snapshot', () => {
    const wrapper = shallow(<InfoScreen />);
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render the the correct components', () => {
    const requiredComponents = [View, Text, Card];
    const wrapper = shallow(<InfoScreen />);
    requiredComponents.forEach((component) => {
      expect(wrapper.find(component)).toHaveLength(1);
    });
  });
});
