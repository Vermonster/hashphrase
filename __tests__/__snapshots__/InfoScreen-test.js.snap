// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoScreen /> should render the same snapshot 1`] = `
<View>
  <withTheme(Card)>
    <withTheme(Text)>
      Test card and text
    </withTheme(Text)>
  </withTheme(Card)>
</View>
`;
